PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['hr.employee']
-    _inherit = ['resource.calendar.attendance']
-    _inherit = ['resource.calendar.leaves']

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_work_entry/models/hr_employee.py
+++ b/addons/hr_work_entry/models/hr_employee.py
@@ -5,7 +5,7 @@ from odoo import models, fields, _
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     has_work_entries = fields.Boolean(compute='_compute_has_work_entries', groups="base.group_system,hr.group_hr_user")
 

--- a/addons/hr_work_entry/models/hr_work_entry.py
+++ b/addons/hr_work_entry/models/hr_work_entry.py
@@ -13,6 +13,7 @@ from odoo.osv import expression
 
 
 class HrWorkEntry(models.Model):
+    _name = 'hr.work.entry'
     _description = 'HR Work Entry'
     _order = 'conflict desc,state,date_start'
 
@@ -244,6 +245,7 @@ class HrWorkEntry(models.Model):
 
 
 class HrWorkEntryType(models.Model):
+    _name = 'hr.work.entry.type'
     _description = 'HR Work Entry Type'
 
     name = fields.Char(required=True, translate=True)

--- a/addons/hr_work_entry/models/resource.py
+++ b/addons/hr_work_entry/models/resource.py
@@ -5,7 +5,7 @@ from odoo import models, fields
 
 
 class ResourceCalendarAttendance(models.Model):
-    _inherit = ['resource.calendar.attendance']
+    _inherit = 'resource.calendar.attendance'
 
     def _default_work_entry_type_id(self):
         return self.env.ref('hr_work_entry.work_entry_type_attendance', raise_if_not_found=False)
@@ -21,7 +21,7 @@ class ResourceCalendarAttendance(models.Model):
 
 
 class ResourceCalendarLeaves(models.Model):
-    _inherit = ['resource.calendar.leaves']
+    _inherit = 'resource.calendar.leaves'
 
     work_entry_type_id = fields.Many2one(
         'hr.work.entry.type', 'Work Entry Type',
